packages/gestalt/src/SideNavigation.js

```diff
@@ -48,6 +41,18 @@ export type Props = {
    * Title for mobile navigation.
    */
   mobileTitle?: string,
+  /**
+   * When passed SideNavigation becomes a collapsible controlled component. If not passed, it stays non-collapsible. See the [collapsible variant](https://gestalt.pinterest.systems/web/sidenavigation#Collapsible) to learn more. This functionality is not supported in mobile.
+   */
+  collapsed?: boolean,
+  /**
+   * Callback fired when the SideNavigation is collapsible and the collapse/expand button is clicked. This functionality is not supported in mobile.
+   */
+  onCollapse?: (boolean) => void,
+  /**
+   * Callback fired when the *collapsed* SideNavigation is hovered and opened in preview mode. This functionality is not supported in mobile.
+   */
+  onPreview?: (boolean) => void,
```

Comment:
This callback had to be introduced so that the (internal) `preview` state is communicated. This is necessary when `header` and `footer` needs to appear the same in both expanded and preview states. Take a look at the `collapsibleHeaderExample.js` file for the usage.

We could also expose `preview` state and make it controlled as well. However, I think it shouldn't be, for the simplicity of the API.

Reply:
@AlbertCarreras We should be opinonated on this. 
The above components have expanded state because they can be exapanded. By default they are NOT expanded. So it makes sense to control that behavior with `expanded` prop. Because expanded state is a change relative to its default state.

SideNavigation, on the other hand, is not an expandible component. It is in expanded state by default. But it can BECOME collapsible. So it makes sense to control that with `collapsed` prop. Because collapsed state is a change relative to its default state.

Reply:
Didn't thought of it that way.  MMM Based on that ... when enabling this feature, isn't the default state "collapsed", then we expanded when interacting. But I guess you mean that SideNavigation needs to be expanded as a default in order to be able to work. But aint the same as Accordion? Accordion default state should be expanded...

haha

idk. 

whatever you decide.

Reply:
> when enabling this feature, isn't the default state "collapsed"

That actually my point. You would enable this feature by adding `collapsed` prop and sidenav will appear as collapsed. :D